storage/storage.go

```diff
@@ -101,9 +102,19 @@ func storeResult(report types.Report, consumerURL string) error {
 	// TODO Move HTTP Post Helper method elsewhere
 	logging.Stderr("[Storage] Posting result to: %s", consumerURL)
 	resp, err := http.Post(consumerURL, "application/json", bytes.NewBuffer(jsonStr))
+
 	if err != nil {
-		logging.Stderr("[Storage] Error posting result, [%s], exiting.", err)
-		os.Exit(1)
+		switch err := err.(type) {
+		case net.Error:
+			if err.Timeout() {
+				os.Stderr.WriteString(fmt.Sprintf("Unable to connect to the HTTP endpoint at %s\nSending scan report to fallback: STDOUT\n", consumerURL))
+				outputResult(report)
+				os.Exit(1)
+			}
+		default:
+			logging.Stderr("[Storage] Error posting result, [%s], exiting.", err)
+			os.Exit(1)
+		}
```

Comment:
I could even see just eliminating the `case` + `default` and always outputting the report on stdout on any failure  this late in the process.